.gitmodules

```diff
@@ -1,6 +1,7 @@
 [submodule "deps/legion"]
 	path = deps/legion
-	url = https://gitlab.com/StanfordLegion/legion.git
+	url = https://github.com/williamberman/legion.git
```

Comment:
Why is this necessary? I'd rather not point master at your fork of legion

Reply:
Yeah don't plan to merge this yet. Still need feedback on if the source change to legion is necessary

Reply:
@jiazhihao These legion changes are the main thing I want you to look at in this PR. @williamberman can provide more explanation about what the changes actually are

Reply:
https://github.com/StanfordLegion/legion/commit/21269d0de541649efccbfe038d6515728e673af9 This is the only remaining legion change. `dom.dim` is volatile when targeting cuda or hip. I put in a const_cast to remove the volatile so it can be passed to deserialize

Reply:
This legion change looks good to me.

Reply:
Ok, put up here https://gitlab.com/StanfordLegion/legion/-/merge_requests/597

I believe this means we'll have to point at legion master instead of legion stable temporarily? Not sure how this works with submodules.

If so, should we make a follow up issue to point back to stable once a new stable release is cut?

Reply:
We use a customized branch of Legion: https://gitlab.com/StanfordLegion/legion/-/tree/flexflow, which is based on Legion's control replication branch. One issue with the master branch of Legion is that it does not support control replication, a key optimization to enable scalability.

Reply:
If the Legion team doesn't want the PR to be merged to the main stream, we can merge it to the `flexflow` branch.

Reply:
@jiazhihao Would it be possible to merge it to the flexflow branch now so we don't have to wait on the legion team? (I imagine you'd have to do that merge as I doubt @williamberman has merge permissions in legion :slightly_smiling_face:)

Reply:
I have merged Will's changes to the flexflow branch of Legion.

Reply:
I will create a PR to update FlexFlow's Legion submodule.

Reply:
Yeah @jiazhihao you probably need to use different branches for different PRs :slightly_smiling_face:

Reply:
Awesome, thank you!

Reply:
> I have merged Will's changes to the flexflow branch of Legion.
It seems to be a bug for the Legion HIP backend. Let's push the fix to the master branch?